### PR TITLE
Change GY ACC Limit

### PR DESCRIPTION
**Pull request by maroqijalil**

### CODE REVIEW

File: src/kansei/imu.cpp [hunk 1]
```diff
@@ -53,11 +56,34 @@ Imu::Imu()
...
+}
+
+void Imu::load_data()
+{
+  std::string file_name =
+    "/home/ichiro/ROS2Project/ros2_ws/src/kansei/data/kansei.json";
+  std::ifstream file(file_name);
+  nlohmann::json imu_data = nlohmann::json::parse(file);
+
+  for (auto &[key, val] : imu_data.items()) {
```
Comment (threeal) [inline src/kansei/imu.cpp:75]: If it's possible, try to use `const auto &`.

File: src/kansei/imu.cpp [hunk 2]
```diff
@@ -53,11 +56,34 @@ Imu::Imu()
...
+
+  load_data();
+}
+
+void Imu::load_data()
+{
+  std::string file_name =
+    "/home/ichiro/ROS2Project/ros2_ws/src/kansei/data/kansei.json";
+  std::ifstream file(file_name);
+  nlohmann::json imu_data = nlohmann::json::parse(file);
```
Comment (threeal) [inline src/kansei/imu.cpp:73]: Put a try catch here to prevent a file failed to be loaded.

File: src/kansei/imu.cpp [hunk 3]
```diff
@@ -53,11 +56,34 @@ Imu::Imu()
...
+
+  for (auto &[key, val] : imu_data.items()) {
+    if (key == "Fallen") {
+      try {
+        val.at("fallen_back_limit").get_to(fallen_back_limit);
+        val.at("fallen_front_limit").get_to(fallen_front_limit);
+        val.at("fallen_right_limit").get_to(fallen_right_limit);
+        val.at("fallen_left_limit").get_to(fallen_left_limit);
+      } catch (nlohmann::json::parse_error & ex) {
```
Comment (threeal) [inline src/kansei/imu.cpp:82]: Use a const reference.

File: include/kansei/imu.hpp [hunk 4]
```diff
@@ -46,9 +45,9 @@ class Imu
 
   void compute_rpy(float gy[3], float acc[3], float seconds);
 
-  float get_roll() {return keisan::rad_to_deg(roll);}
-  float get_pitch() {return keisan::rad_to_deg(pitch);}
-  float get_yaw() {return keisan::rad_to_deg(yaw);}
+  float get_roll() {return roll;}
+  float get_pitch() {return pitch;}
+  float get_yaw() {return yaw;}
```
Comment (threeal) [inline include/kansei/imu.hpp:50]: Use `const type & function_name() const` for a getter that return a member variable of a class. Do this to the other function too.

File: src/kansei/imu.cpp [hunk 5]
```diff
@@ -53,11 +56,34 @@ Imu::Imu()
...
   fb_gyro_center = 512.0;
 
-  fallen_back_limit = 390.0;
-  fallen_front_limit = 590.0;
-  fallen_right_limit = 390.0;
-  fallen_left_limit = 610.0;
+  fallen_back_limit = 620.0;
+  fallen_front_limit = 445.0;
+  fallen_right_limit = 570.0;
+  fallen_left_limit = 400.0;
```
Comment (threeal) [inline src/kansei/imu.cpp:62]: Put default values in the delegate (`: foo(0.0)`).

**Review by threeal — CHANGES_REQUESTED** [review]
See individual comment for the requested changes.

File: data/kansei.json [hunk 6]
```diff
@@ -4,5 +4,10 @@
     "fallen_front_limit": 445.0,
     "fallen_right_limit": 570.0,
     "fallen_left_limit": 400.0
+  },
+  "Imu" {
```
Comment (threeal) [inline data/kansei.json:8]: Don't mix naming case in configuration file, use either classic `snake_case` or javascript's `camelCase`.

**Review by threeal — CHANGES_REQUESTED** [review]
Just a minor change request.

**Review by threeal — APPROVED** [review]
LGTM

**Review by finesaaa — APPROVED** [review]
Good

**Pull request merged**